sdk/core/azure-core/tests/test_messaging_cloud_event.py

```diff
@@ -146,6 +144,108 @@ def test_cloud_custom_dict_with_extensions():
     assert event.time.microsecond == 539861
     assert event.extensions == {"ext1": "example", "ext2": "example2"}
 
+def test_cloud_custom_dict_ms_precision_is_gt_six():
+    cloud_custom_dict_with_extensions = {
+        "id":"de0fd76c-4ef4-4dfb-ab3a-8f24a307e033",
+        "source":"https://egtest.dev/cloudcustomevent",
+        "data":{"team": "event grid squad"},
+        "type":"Azure.Sdk.Sample",
+        "time":"2021-02-18T20:18:10.539861122+00:00",
+        "specversion":"1.0",
+    }
+    event = CloudEvent.from_dict(cloud_custom_dict_with_extensions)
+    assert event.data == {"team": "event grid squad"}
+    assert event.__class__ == CloudEvent
+    assert event.time.month == 2
+    assert event.time.day == 18
+    assert event.time.hour == 20
+    assert event.time.microsecond == 539861
```

Comment:
qq: so the test is checking that precision is greater than 6 decimal places, right? In that case, wouldn't you want to check that the ms `==539861122`? or am I getting mixed up?

Reply:
Is testing that when we have more than 6 decimal places, it is trimmed to 6 - python date time objects ms precision has a limit of 6 digits